runtime/bindings/python/src/pyopenvino/graph/passes/manager.cpp

```diff
@@ -9,22 +9,59 @@
 
 #include "openvino/pass/constant_folding.hpp"
 #include "openvino/pass/pass.hpp"
+#include "openvino/pass/serialize.hpp"
 #include "openvino/pass/validate.hpp"
 #include "pyopenvino/graph/passes/manager.hpp"
 
 namespace py = pybind11;
 
+
+// TODO:
+// better expections
+// better docs
 namespace {
 class ManagerWrapper : public ov::pass::Manager {
 public:
     ManagerWrapper() {}
     ~ManagerWrapper() {}
-    void register_pass(std::string pass_name) {
+
+    // void register_pass(std::string pass_name, py::args args, const py::kwargs& kwargs) {
+    //     if (pass_name == "ConstantFolding")
+    //         push_pass<ov::pass::ConstantFolding>();
+
+    //     if (m_per_pass_validation)
+    //         push_pass<ov::pass::Validate>();
+
+    //     if (pass_name == "Serialize") {
+    //         const auto args_size = args.size();
+    //         const auto kwargs_size = kwargs.size();
+    //         if ((args_size > 0) || (kwargs_size > 0)) {
+    //             if (args_size + kwargs_size > 2) {
+    //                 throw ov::Exception("Invoked with too many arguments! The possible args are: xml_path and
+    //                 bin_path");
+    //             }
+    //             const auto xml_path = kwargs.contains("xml_path") ? py::cast<std::string>(kwargs["xml_path"]) : py::cast<std::string>(args[0]);
+    //             const auto bin_path = kwargs.contains("bin_path") ? py::cast<std::string>(kwargs["bin_path"]) : py::cast<std::string>(args[1]);
+    //             push_pass<ov::pass::Serialize>(xml_path, bin_path, Version::UNSPECIFIED);
+    //         } else {
+    //              throw ov::Exception("No required paths specified for saving .xml and .bin files!");
+    //         }
+    //     }
```

Comment:
This looks very good. But what if there is only one additional argument?

Reply:
good point, there always has to be 2 additional args for paths, so I guess the simple change to ` if( args_size + kwargs_size == 2) { ... } should be sufficient`

Reply:
I vote for this solution.
BTW, what about the option if pass paths as tuple?
Like we discussed: `.register_pass("Serialize", output_file=(xml_path, bin_path))`

Reply:
I think it can be resolved by extending if statement for checking if arg_size + kwarg_size == 1 and then checking if kwargs contains output_file and casting it to the std::pair or tuple and grabbing xml and bin paths from it

Reply:
so the user will have a choice to either write `.register_pass("Serialize", xml_path=xml_path, bin_path=bin_path))` or `.register_pass("Serialize", output_file=(xml_path, bin_path))`

Reply:
Honestly I leave it up to you to pick one method. For me both are fine:

1. creating two separate function handled correctly based on pybind's args resolving.
2. one function with args/kwargs dispatching.

Just please drop `> 4` for files, it should be handled by underlaying logic if at all - not out binding code.